onnx_tool/graph.py

```diff
@@ -521,12 +521,59 @@ def remove_node(self, nodename):
                 self.producedby[o].remove(nodename)
                 if len(self.producedby[o]) == 0:
                     self.producedby.pop(o)
+
+                if o in self.output:
+                    self.output.remove(o)
```

Comment:
After removing it from self.output, it should append new tensors to self.output.

Reply:
Not sure how it should be handled, on one hand this changes previous behavior, on other previous implementation leaves us with just dangling output which one are not very useful by itself.
Also there is may be case when multiple tensors stream in removed nodes, should we add new output for each?

Reply:
Back from vacation. Here is the simple situation: the graph has one output and the removal removes this output. The new output tensor should be the input tensor of the subtree. If the removed has no output tensor, it's not a completed onnx graph.

Reply:
Sure, yet dangling output not very useful too. Maybe we may introduce one more method `remove_dangling_outputs` to be explicitly called by user?

Reply:
Let's make your PR simple to merge. Could you show me your use case? Maybe a test model and some test codes. I want to make sure the generated model is a valid ONNX model. If there is no problem with your use case, let me merge your PR first. Fix the failed cases when they are issued.

Reply:
@Mikhael-Danilov If sharing model is not a feasible way for you, you can draw a scheme about how to remove the subtree from the graph. Anyway, the least effort to show that the code works fine for your situation.

Reply:
In my case it looks like:

`model = YOLO(modelPath)
model.export(format="onnx",imgsz=imgsz)

model = onnx_tool.Model(modelPath.replace(".pt", ".onnx"))

graph = model.graph
graph.shape_infer()

...

graph.add_dump_tensors(["/model.22/cv3.2/cv3.2.2/Conv_output_0"])
graph.remove_subtree("/model.22/Concat_2")
graph.remove_dangling_nodes()

...

model.save_model(onnxPath)`

with yolov8n as model.

So:
* I define new output
* Cut the reset
* Cleanup

Reply:
I see, you manually add a new output tensor. Before, we talked about the automatic method for the new output tensor. The manual way is also okay for me.